src/sensors/src/Sensors.cpp

```diff
@@ -931,7 +926,7 @@ bool SensorsMeasurements::getMeasurement(const SensorType &sensor_type, const un
 
     if( sensor_type == GYROSCOPE )
     {
-        if( sensor_index < this->pimpl->GyroscopeMeasurements.size() )
+        if( static_cast<std::size_t>(sensor_index) < this->pimpl->GyroscopeMeasurements.size() )
```

Comment:
Why the ``static_cast``, what type ``size()`` is returning?

Reply:
size() returns std::size_t (i.e. unsigned) for consistency with STL containers, while `sensor_index` is `ptrdiff_t` (i.e. signed), because it can be contained a invalid index (i.e. -1)
The static_cast ensures that we are comparing the same types.

Reply:
Ah ok, this answers also my doubt on ``ptrdiff_t``, clear!